pkg/codegen/codegen.go

```diff
@@ -400,16 +429,15 @@ func (b *Builder) WriteTemplate(ctx context.Context, filePath,
 
 	warnings := make([]string, 0)
 
-	if b.HasDeviations(ctx, filePath) {
-		warnings = append(warnings, fmt.Sprintf("SKIPPED: '%s' had deviations and will not be re-generated", filePath))
-		return warnings, nil
-	}
-
 	templates, err := b.renderTemplate(filePath, contents, args)
 	if err != nil {
 		return nil, err
 	}
 
+	// TODO(jaredallard): I think this implementation will break if
+	// a multi-file template has blocks with different contents. Need
+	// to look into this. Possible we'll need to pre-render to
+	// determine which files we're writing to to populate args.
```

Comment:
This TODO is very valid, thinking through this. It's going to cause issues with any files that generate more files, e.g. the command processor.

We'll need to bring the actual "get blocks" code into this loop as well. This is slightly problematic because we don't know what files we're going to write to until _after_ we execute the template. I'm going to need to think about this more. The naive solution would be to execute it once, read the files, then you'd know, but that probably still won't solve it because the template itself is the one iterating over the files and writing the contents,

Instead, I think the solution is to have `stencil.ApplyTemplate` do the blocks logic. This will require a signature change to have it know which file it's going to be written to (right now it's piped to `stencil.InstallFile`), but this would allow us to support that case.

For context about what I'm going on about/a sample file, see the pkg.go.dev docs: https://pkg.go.dev/github.com/getoutreach/stencil/pkg/functions#Stencil.InstallFile